src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -311,7 +307,7 @@ internal SourceText TryReadFileContent(CommandLineSourceFile file, IList<Diagnos
                 var directory = Path.GetDirectoryName(normalizedPath) ?? normalizedPath;
```

Comment:
It looks like `normalizedPath` is never null here, or `fileContent` would have been null. Consider asserting and then removing the check below.